File: Java/opentraceability/src/main/java/opentraceability/utility/DoubleExtensions.java
package opentraceability.utility;

public class DoubleExtensions {
    public static double Round(double number) {
        double roundedValue = number;
        String strVal = String.format("e12", number);
        roundedValue = Double.parseDouble(strVal);
        return roundedValue;
    }

    public static Double Round(Double number) {
        if (number == null) {
            return null;
        }

        double roundedValue = number;
        String strVal = String.format("e12", number);
        roundedValue = Double.parseDouble(strVal);
        return roundedValue;
    }
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/utility/doubleextensions.kt
package utility

class DoubleExtensions {
    companion object {
        fun Round(number: Double): Double {
            var roundedValue: Double = number
            var strVal: String = String.format("e12", number)
            roundedValue = strVal.toDouble()
            return roundedValue
        }

        fun Round(number: Double?): Double? {

            if (number == null){
                return number
            }

            var roundedValue: Double? = number
            var strVal: String = String.format("e12", number)
            roundedValue = strVal.toDouble()
            return roundedValue
        }
    }
}
